plugin/metrics/prometheus/factory.go

```diff
@@ -47,19 +47,23 @@ func (f *Factory) InitFromViper(v *viper.Viper, logger *zap.Logger) {
 }
 
 // Initialize implements storage.MetricsFactory.
-func (f *Factory) Initialize(logger *zap.Logger) error {
-	f.logger = logger
+func (f *Factory) Initialize(telset telemetry.Settings) error {
+	f.telset = telset
 	return nil
 }
 
 // CreateMetricsReader implements storage.MetricsFactory.
 func (f *Factory) CreateMetricsReader() (metricsstore.Reader, error) {
-	return prometheusstore.NewMetricsReader(f.options.Configuration, f.logger, f.tracer)
+	mr, err := prometheusstore.NewMetricsReader(f.options.Configuration, f.telset.Logger, f.telset.TracerProvider)
+	if err != nil {
+		return mr, err
```

Comment:
```suggestion
		return nil, err
```

Reply:
can we add a test?

Reply:
@yurishkuro We'll need to dip into the implementation of `NewMetricsReader ` to force an error here. Is that fine? Also, since we're just decorating the reader, do we want to force returning a nil if there is an error? My thinking was that we just pass along whatever it is we get.

Reply:
You can simulate the error easily by passing a TLS config with "foobar" for some of the certificates.

It is convention to return `nil, err` in case of errors. It's probably what you would get from the factory already, but when you return `mr, err` you are returning a typed nil, so the nil check may actually fail in the caller.

Reply:
@yurishkuro sounds good! i'll also go back and do the same for the other factories in a follow-up PR